core/src/main/scala/org/http4s/parser/UriHeaderParser.scala

```diff
@@ -0,0 +1,19 @@
+package org.http4s.parser
+
+import java.nio.charset.StandardCharsets
+
+import org.http4s.{Header, Uri}
+import org.http4s.internal.parboiled2.Rule1
+
+abstract class UriHeaderParser[A <: Header](value: String) extends Http4sHeaderParser[A](value) with Rfc3986Parser {
+
+  override def charset: java.nio.charset.Charset = StandardCharsets.ISO_8859_1
```

Comment:
I'm sorry I merged after @rossabaker approval. This is taken from the original used in `Location`, The link is interesting though I believe that international characters are valid on at least the domain name, like indicated at 

http://unicode.org/faq/idn.html